parsers/EIA.py

```diff
@@ -418,7 +418,7 @@ def fetch_production_mix(zone_key, session=None, target_datetime=None, logger=No
                 0 > point['value'] >= negative_threshold:
                 point['value'] = 0
 
-            if type == 'hydro' and point['value'] < 0:
+            if type == 'hydro' and point['value'] and point['value'] < 0:
```

Comment:
Is it possible to distinguish, in the raw data, between pumped hydro and run-of-the-river hydro?
Right now it seems that we interpret positive values as run-of-the-river, and negative values as storage, which causes us to never register the times at which pumped hydro generates electricity (we incorrectly interpret that situation as non-pumped hydro generation instead)

Reply:
Hey, sorry for the late reply! To be honest, I'm super new to the project (this is my first PR) so I have no idea. I'll look into it when I get some time this week. 👍

Reply:
Hey @corradio , as far as I can tell it's not possible to distinguish in the raw data, which just contains the "Net generation from hydro for x region".

Here's an example request:
https://api.eia.gov/series/?api_key=<EIA_API_KEY>&series_id=EBA.TEN-ALL.NG.WAT.H&num=24

Does this project distinguish between run-of-the-river and pumped hydro anywhere? From what I can see, negative values are interpreted as storage and positive values are interpreted as general "hydro".

Reply:
Hi @jakedorne,
Thanks for investigating! This project indeed distinguishes between pumped hydro (which in electricityMap world is the `hydro` key under `storage`) and run-of-the-river (which is the `hydro` key under `production`).
If we only have access to the net generation, then I think we should put the net under `storage` (as it can become negative).
Does the EIA API allow us to separate run-of-the-river hydro and pumped hydro?

Reply:
Hey @corradio, sorry, I think I misunderstood your original comment - I thought you meant there were a total of 3 distinctions (pumped, run-of-the-river, and storage).

I'm struggling to find any way to distinguish between pumped and run-of-the-river generation.

If there is no way to distinguish, I'm happy to edit the code to add the net generation under storage, but does that not mean we'd have the opposite problem and be interpreting all generation as as pumped?

Just to confirm: If I can find a way to distinguish, should all pumped generation be added to production and all run-of-the-river be added to storage?

Again, sorry for the late reply. Busy times. 😞

Reply:
> If there is no way to distinguish, I'm happy to edit the code to add the net generation under storage, but does that not mean we'd have the opposite problem and be interpreting all generation as as pumped?

Correct. I suggest we leave it like that for now.

> Just to confirm: If I can find a way to distinguish, should all pumped generation be added to production and all run-of-the-river be added to storage?

If we can find a way to distinguish, pumped generation (which can become negative) should be added to storage, and run-of-the-river (which can never be negative) added to production.

> Again, sorry for the late reply. Busy times. 😞

No worries! Thanks again